src/cargo/core/shell.rs

```diff
@@ -447,6 +460,10 @@ mod imp {
 
     pub(super) use super::default_err_erase_line as err_erase_line;
 
+    pub fn accurate_stderr_width() -> Option<usize> {
+        None
```

Comment:
Would it be possible to implement the enum mention in [here](https://github.com/rust-lang/cargo/pull/7315#discussion_r320539231), so that this works a little better on Windows?

Reply:
I haven't done this because I've removed the relevant changes from the PR in response to one of your other comments.

Reply:
Sure:

* `notty`: would be when `err_width` currently returns `None`.
* `known`: would be when `stderr_width` returns Some.
* `unknown`: would be the [mintty](https://github.com/rust-lang/cargo/blob/40550b9ea04f2b991202e11b43f46086424a7cfb/src/cargo/core/shell.rs#L477-L487) case, and would require special handling.  A different name could be "width guess" since it is just taking a guess.  In this situation, we still want to use the conservative "60" estimate for the progress bar, but I'm guessing for `-Zterminal-width` we should probably just treat it the same as `notty` (use infinite width, no truncation).  From the output I looked at, clipping at 60 columns looked pretty terrible.

Reply:
I've implemented this - let me know if it was what you were intending.